src/tests/class_tests/openms/source/CoarseIsotopeDistribution_test.cpp

```diff
@@ -291,6 +291,64 @@ START_SECTION(IsotopeDitribution CoarseIsotopePatternGenerator::estimateFromPept
 }
 END_SECTION
 
+START_SECTION(IsotopeDitribution CoarseIsotopePatternGenerator::approximateFromPeptideWeight(double mass, int num_peaks))
+{
+  std::vector<float> masses_to_test = {20, 300, 1000, 2500};
+  for (auto mass = masses_to_test.begin(); mass != masses_to_test.end(); ++mass)
+  {
+    IsotopeDistribution approximation = CoarseIsotopePatternGenerator::approximateFromPeptideWeight(*mass);
+    solver->setMaxIsotope(approximation.size());
+    IsotopeDistribution coarse_truth = solver->estimateFromPeptideWeight(*mass);
+
+    // compute KL divergence (Sum over all x: P(x) * log(P(x) / Q(x)), where P is a distribution and Q its approximation
+    double KL = 0;
+    double sum = 0.0;
+    for (UInt peak = 0; peak != approximation.size() && peak != coarse_truth.size(); ++peak) // coarse_truth.size() is 18, although approximation.size() = 20 for mass = 20
+    {
+      double Px = coarse_truth[peak].getIntensity();
+      double Qx = approximation[peak].getIntensity();
+      sum += Qx;
+      if (Px != 0) // KL is 0 for Px = 0
+      {
+        KL += Px * log(Px / Qx);
+      }
+    }
+
+    TEST_REAL_SIMILAR(sum, 1.0);
+    TEST_EQUAL(0 < KL && KL < 0.05, true);
+  }
+}
+END_SECTION
+
+START_SECTION(IsotopeDitribution CoarseIsotopePatternGenerator::approximateIntensities(double mass, int num_peaks))
+{
+  std::vector<Int> masses_to_test = {20, 300, 1000, 2500};
+  for (auto mass = masses_to_test.begin(); mass != masses_to_test.end(); ++mass)
+  {
+    std::vector<double_t> approximation = CoarseIsotopePatternGenerator::approximateIntensities(*mass);
```

Comment:
`double_t` ?
Please don't....

Reply:
I can change... gitpod is just click away :)

Reply:
Was not aware that is an issue... Thanks @timosachsenberg for fixing!

Reply:
Its not an issue per se, but it violates "Do not write surprising code" :) and it potentially makes testing a lot harder (or fail), if some platform or someone decides to set this strange FP.. flag, which changes the underlying type of double.